src/indieweb_utils/posts/discovery.py

```diff
@@ -97,10 +98,10 @@ def discover_original_post(posse_permalink: str):
                 if original_post_url:
                     return original_post_url
 
-    return None
+    return ""
 
 
-def discover_author(url, page_contents=None):
+def discover_author(url: str, page_contents=None) -> dict:
```

Comment:
For a different PR, but I wonder if we couldn't make this return our Author object?

Reply:
The h-card microformat has a lot of potential values, including one that is nested: https://microformats.org/wiki/h-card

I think typeclasses are easier to read but it's worth discussing this since we will have to introduce quite a bit of repetitive code in the return function to accomodate for all the values in a h-card.

Reply:
We shouldn't end up with too much redundant code. My thinking is that

1. We should have h-card properties listed in a constants file (just the common 4 or 5 properties is fine to start). We'd also have a list of "valid properties".
2. Loop through the passed properties and store the valid ones in a dict
3. Use that dictionary to populate the dataclass

```python
import dataclasses

@dataclasses.dataclass
class Author:
    ...
    p_job_title: str = dataclasses.field(default="")
    # Or maybe it Optional[str] and set default to None?


def discover_author(..):
    author_properties ={key: value for key, value in properties.values() if key in valid_author_properties} 
    ...
    # pass the dict as kwargs to the class
    return Author(**author_properties)  
```

I'll try and get a PR together for it once this merges.

Reply:
Thank you @jamesvandyne! I pushed all of your feedback in #e006147. Pending your approval, we can merge this PR into the main branch so we can get to work on the h-card / authorship representation change.